data-pipeline-flink/telemetry-extractor/src/main/scala/org/sunbird/dp/functions/DeduplicationFunction.scala

```diff
@@ -0,0 +1,79 @@
+package org.sunbird.dp.functions
+
+import java.util
+
+import com.google.gson.Gson
+import org.apache.flink.api.common.typeinfo.TypeInformation
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.streaming.api.functions.ProcessFunction
+import org.apache.flink.streaming.api.scala.OutputTag
+import org.apache.flink.util.Collector
+import org.slf4j.LoggerFactory
+import org.sunbird.dp.cache.{DedupEngine, RedisConnect}
+import org.sunbird.dp.task.ExtractionConfig
+
+class DeduplicationFunction(config: ExtractionConfig, @transient var dedupEngine: DedupEngine = null)(implicit val eventTypeInfo: TypeInformation[util.Map[String, AnyRef]])
+  extends ProcessFunction[util.Map[String, AnyRef], util.Map[String, AnyRef]] {
+
+  private[this] val logger = LoggerFactory.getLogger(classOf[DeduplicationFunction])
+
+  /**
+   * Tag to hold all duplicate batch events based on the msgId
+   * (Note: De-dup validation only if batch event has "msgId" )
+   */
+  lazy val duplicateEventOutput: OutputTag[util.Map[String, AnyRef]] = new OutputTag[util.Map[String, AnyRef]](id = "duplicate-events")
```

Comment:
This can change to lazy val duplicateEventOutput: OutputTag[util.Map[String, AnyRef]] = OutputTag[util.Map[String, AnyRef]](id = "duplicate-events"). No need of "new" keyword in scala